### PR TITLE
SSA Improvements

### LINKED ISSUE #3882: SSA improvements
- [ ] Result tabs along with warning checkboxes should be shown on analysis running not on compilation
- [ ] If all tools are unchecked, in result section, result is shown but not the tabs
- [ ] Show warnings and hide warnings labels can be confusing for some users
- [ ] Position on Solhint warnings should be improved
- [ ] Text on badge tooltip can have count

### PR DESCRIPTION
**Pull request by joeizang**
fixes #3882 

- [x] Result tabs along with warning checkboxes should be shown on analysis running not on compilation
- [x] If all tools are unchecked, in result section, result is shown but not the tabs
- [x] Show warnings and hide warnings labels can be confusing for some users
- [x] Position on Solhint warnings should be improved
- [x] Text on badge tooltip can have count

### CODE REVIEW

**Review by Aniket-Engg — COMMENTED** [review]
If I uncheck all boxes, and click on `show errors only` checkbox, badge shows message with count
Screenshot 2023-07-05 at 6 55 58 PM

File: libs/remix-ui/static-analyser/src/lib/ErrorRenderer.tsx [hunk 1]
```diff
@@ -1,71 +1,97 @@
...
-    if (!msg.includes(':')) return { errLine: -1, errCol: -1, errFile: msg }
+    if (!msg.includes(":")) return { errLine: -1, errCol: -1, errFile: msg };
 
     // extract line / column
-    let position = msg.match(/^(.*?):([0-9]*?):([0-9]*?)?/)
-    result.errLine = position ? parseInt(position[2]) - 1 : -1
-    result.errCol = position ? parseInt(position[3]) : -1
+    let position = msg.match(/^(.*?):([0-9]*?):([0-9]*?)?/);
+    result.errLine = position ? parseInt(position[2]) - 1 : -1;
+    result.errCol = position ? parseInt(position[3]) : -1;
```
Comment (Aniket-Engg) [inline libs/remix-ui/static-analyser/src/lib/ErrorRenderer.tsx:24]: If you find it fine, I think we can remove all semicolons

File: libs/remix-ui/static-analyser/src/lib/actions/staticAnalysisActions.ts [hunk 2]
```diff
@@ -198,12 +207,13 @@ slitherEnabled: boolean, setStartAnalysis: React.Dispatch<React.SetStateAction<b
           props.analysisModule.call('terminal', 'log', { type: 'error', value: '[Slither Analysis]: Error occured! See remixd console for details.' })
           showWarnings(warningMessage, 'warningModuleName')
         }
-      } else showWarnings(warningMessage, 'warningModuleName')
+      } else //showWarnings(warningMessage, 'warningModuleName')
```
Comment (Aniket-Engg) [inline libs/remix-ui/static-analyser/src/lib/actions/staticAnalysisActions.ts:210]: remove commented code

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 3]
```diff
@@ -436,124 +437,144 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+      }
     props.event.trigger("staticAnaysisWarning", [
       slitherTotal + ssaTotal + hintsTotal === 0
         ? -1
         : slitherTotal + ssaTotal + hintsTotal,
-    ]);
+    ])
   }, [solhintEnabled, basicEnabled, slitherEnabled, showSlither])
 
+  useEffect(() => {
```
Comment (Aniket-Engg) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:531]: What is this for?

Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:531]: just missed removing it

File: libs/remix-ui/static-analyser/src/lib/actions/staticAnalysisActions.ts (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/actions/staticAnalysisActions.ts:210]: sure

**Pull request merged**